tests/components/blebox/test_light.py

```diff
@@ -0,0 +1,597 @@
+        feature_mock.brightness = 0xAB
+        feature_mock.is_on = True
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+
+    await async_setup_entity(hass, config, entity_id)
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_ON
+    assert state.attributes[ATTR_BRIGHTNESS] == 0xAB
+
+
+async def test_wlightbox_s_on(wlightbox_s, hass, config):
+    """Test light on."""
+
+    feature_mock, entity_id = wlightbox_s
+
+    def initial_update():
+        feature_mock.is_on = False
+        feature_mock.sensible_on_value = 254
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+    feature_mock.async_update = AsyncMock()
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_OFF
+
+    def turn_on(brightness):
+        assert brightness == 254
+        feature_mock.brightness = 254  # on
+        feature_mock.is_on = True  # on
+
+    feature_mock.async_on = AsyncMock(side_effect=turn_on)
+    await hass.services.async_call(
+        "light", SERVICE_TURN_ON, {"entity_id": entity_id}, blocking=True,
+    )
+
+    state = hass.states.get(entity_id)
+    assert state.attributes[ATTR_BRIGHTNESS] == 254
+    assert state.state == STATE_ON
+
+
+@pytest.fixture(name="wlightbox")
+def wlightbox_fixture():
+    """Return a default light entity mock."""
+
+    feature = mock_feature(
+        "lights",
+        blebox_uniapi.light.Light,
+        unique_id="BleBox-wLightBox-1afe34e750b8-color",
+        full_name="wLightBox-color",
+        device_class=None,
+        is_on=None,
+        supports_color=True,
+        supports_white=True,
+        white_value=None,
+        rgbw_hex=None,
+    )
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My wLightBox")
+    type(product).model = PropertyMock(return_value="wLightBox")
+    return (feature, "light.wlightbox_color")
+
+
+async def test_wlightbox_init(wlightbox, hass, config):
+    """Test cover default state."""
+
+    _, entity_id = wlightbox
+    entry = await async_setup_entity(hass, config, entity_id)
+    assert entry.unique_id == "BleBox-wLightBox-1afe34e750b8-color"
+
+    state = hass.states.get(entity_id)
+    assert state.name == "wLightBox-color"
+
+    supported_features = state.attributes[ATTR_SUPPORTED_FEATURES]
+    assert supported_features & SUPPORT_WHITE_VALUE
+    assert supported_features & SUPPORT_COLOR
+
+    assert ATTR_WHITE_VALUE not in state.attributes
+    assert ATTR_HS_COLOR not in state.attributes
+    assert ATTR_BRIGHTNESS not in state.attributes
+    assert state.state == STATE_OFF
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+
+    assert device.name == "My wLightBox"
+    assert device.identifiers == {("blebox", "abcd0123ef5678")}
+    assert device.manufacturer == "BleBox"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)